### PR TITLE
JSHint cleanup

### PR DESCRIPTION
**Pull request by XhmikosR**
BTW @cgiffard, I could set up JSCS and or ESLint to make things more consistent across the codebase.

Also, I was thinking, maybe we should just move JSHint in package.json script section?

### CODE REVIEW

**Comment by XhmikosR** [comment]
BTW @cgiffard, I could set up JSCS and or ESLint to make things more consistent across the codebase.

Also, I was thinking, maybe we should just move JSHint in package.json script section?

**Comment by cgiffard** [comment]
Yep. The test encapsulation of JSHint dates to a time where JSHint had a really rubbish CLI. Now, it's fine. :)

**Comment by XhmikosR** [comment]
All right, I'll update the PR and ping you.

**Comment by XhmikosR** [comment]
@cgiffard: updated. Let me know if you need something else.

PS.
> BTW @cgiffard, I could set up JSCS and or ESLint to make things more consistent across the codebase.

**Comment by cgiffard** [comment]
There are a couple of tiny stylistic things that I wanna tweak with the array literal syntax, but other than that, looks good! I'm really tired so I'll do a manual merge in the morning. :)

**Comment by cgiffard** [comment]
I've used JSCS and it'd be cool to get it integrated to ensure contributors' style is correct. I might have a crack at it tomorrow (unless you get to it first!)

**Comment by XhmikosR** [comment]
> There are a couple of tiny stylistic things that I wanna tweak with the array literal syntax, but other than that, looks good! I'm really tired so I'll do a manual merge in the morning. :)

Do you mean in this patch or in general?

> I've used JSCS and it'd be cool to get it integrated to ensure contributors' style is correct. I might have a crack at it tomorrow (unless you get to it first!)

Should be easy to add it so I might have some time to make a PR later, after this is merged.

**Comment by XhmikosR** [comment]
Made a few more tweaks.

**Comment by XhmikosR** [comment]
Allright, I'm done for now. I used the JSHint rules you see in this PR and ESLint. Would be nice to have this automated but I'll leave it up to you after this is merged.

File: example/quickcrawl-example.js [hunk 1]
```diff
@@ -4,9 +4,9 @@
 var Crawler = require("../lib");
 
 Crawler.crawl("http://deewr.gov.au/")
-	.on("fetchstart",function(queueItem){
-		console.log("Starting request for:",queueItem.url);
+	.on("fetchstart", function(queueItem) {
+		console.log("Starting request for:", queueItem.url);
```
Comment (XhmikosR) [inline example/quickcrawl-example.js]: Huh? It's defined just fine.

File: example/savetodisk.js [hunk 2]
```diff
@@ -1,73 +1,75 @@
...
+			// Else, recursively create dir using node-fs, then write file
+			else {
+				fs.mkdir(dirname, 0755, true, function () {
+					fs.writeFile(filepath, responseBuffer, function () {});
+				});
+			}
+
+		});
+
+		console.log("I just received %s (%d bytes)", queueItem.url, responseBuffer.length);
```
Comment (XhmikosR) [inline example/savetodisk.js]: Please stop spamming me with console is not defined. It is defined. As for the line, it's something I don't care in this PR.

File: example/phantom-example/index.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
+/* jshint browser:true */
+
 var phantomAPI	= require("phantom"),
-	Crawler 	= require("simplecrawler"),
-	async		= require("async"),
-	colors		= require("colors"),
-	fs			= require("fs"),
-	phantomBin	= __dirname + "/node_modules/.bin/phantomjs";
+	Crawler		= require("simplecrawler"),
+	path		= require("path"),
```
Comment (XhmikosR) [inline example/phantom-example/index.js]: You clearly have no idea what you are talking about.

**Comment by XhmikosR** [comment]
@cgiffard: please step up and make that @houndci guy go away from this PR. Not only he is wrong, he keeps spamming the PR with useless comments.

Otherwise I will just close this and be done with it.

**Comment by XhmikosR** [comment]
@cgiffard: bump for feedback.

**Comment by cgiffard** [comment]
Cleaning up old PRs — going to do a bit of a code cleanup soon, so I figured I'd close this one for now.

**Comment by XhmikosR** [comment]
I could rebase if needed since I've done the work already.

**Comment by cgiffard** [comment]
Sure, I did a quick review and it looks good. If you're happy to rebase, I'll merge!

**Comment by cgiffard** [comment]
There's definitely style stuff there that could evolve, but my feelings on this have mellowed — I think it's more important to be consistent than right.

**Comment by XhmikosR** [comment]
Rebased this but apparently you have closed it.

**Comment by cgiffard** [comment]
I tried to reopen, but it looks like you might need to file a new PR... it won't let me. :-/

Sorry, mate.

**Comment by XhmikosR** [comment]
Ah, OK, no worries. Opened #163 .

**Pull request closed without merging**